crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -118,6 +118,38 @@ where
         .map(|ext| ext.encode())
     }
 
+    fn derive_domain_set_code_extrinsic(
+        &self,
+        consensus_block_hash: H256,
+        domain_id: DomainId,
+    ) -> Option<SetCodeExtrinsic> {
```

Comment:
Somehow github allow to comment on same thread here - https://github.com/subspace/subspace/pull/2132#discussion_r1367296054

I made it into two variants since outer option is when derivation itself failed for some other reason. Making this into 3 variants seems very awkard to me and not consistent with other functions we have. 2 variants did indeed improved some readability a bit. Let me know if you think otherwise.